torch/csrc/jit/script/init.cpp

```diff
@@ -529,8 +529,38 @@ void initJitScriptBindings(PyObject* module) {
           "_register_attribute",
           [](Module& self, std::string name, TypePtr type, py::object value) {
             auto unshaped = unshapedType(type);
-            self.register_attribute(
-                name, unshaped, toIValue(std::move(value), type));
+            if (type->cast<InterfaceType>()) {
```

Comment:
This logic is not in the right place.

The `toIValue(an_class, the_interface_type)` should handle the conversion of the class into an IValue. There are other places where toIValue is called and they should work with interfaces. This only makes it work for attribute assignment.